filters/builtin/compress.go

```diff
@@ -104,47 +119,60 @@ func (e encodings) Swap(i, j int)      { e[i], e[j] = e[j], e[i] }
 //
 // It is possible to control the compression level, by setting it as the first
 // filter argument, in front of the MIME types. The default compression level is
-// best-speed. The possible values are integers between 0 and 9 (inclusive), where
-// 0 means no-compression, 1 means best-speed and 9 means best-compression.
+// best-speed. The possible values are integers between 0 and 11 (inclusive), where
+// 0 means no-compression, 1 means best-speed and 11 means best-compression.
 // Example:
 //
 // 	* -> compress(9, "image/tiff") -> "https://www.example.org"
 //
 // The filter also checks the incoming request, if it accepts the supported
 // encodings, explicitly stated in the Accept-Encoding header. The filter currently
-// supports gzip and deflate. It does not assume that the client accepts any
-// encoding if the Accept-Encoding header is not set. It ignores * in the
+// supports brotli, gzip and deflate. It does not assume that the client accepts any
+// encodingPriority if the Accept-Encoding header is not set. It ignores * in the
 // Accept-Encoding header.
 //
+// Supported encodings are prioritized on:
+// - quality value if provided by client
+// - server side priority (encodingPriority) otherwise
+//
 // When compressing the response, it updates the response header. It deletes the
 // the Content-Length value triggering the proxy to always return the response
-// with chunked transfer encoding, sets the Content-Encoding to the selected
-// encoding and sets the Vary: Accept-Encoding header, if missing.
+// with chunked transfer encodingPriority, sets the Content-Encoding to the selected
+// encodingPriority and sets the Vary: Accept-Encoding header, if missing.
 //
 // The compression happens in a streaming way, using only a small internal buffer.
 //
-func NewCompress() filters.Spec { return &compress{} }
+func NewCompress() filters.Spec { return NewCompressWithEncodings(supportedEncodings) }
+
+func NewCompressWithEncodings(encoding []string) filters.Spec {
```

Comment:
Since we are introducing options, lets make it future proof by using `Options` struct and `NewCompressWithOptions`, otherwise we would need to invent a new constructor for each new option

Reply:
good point, updated